--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   pyproject.toml
	modified:   source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py
	modified:   source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/velocity_env_cfg.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/pyproject.toml b/pyproject.toml
index aa557401..f67955bb 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -75,7 +75,7 @@ exclude = [
     ".vscode",
 ]
 
-typeCheckingMode = "basic"
+typeCheckingMode = "off"
 pythonVersion = "3.11"
 pythonPlatform = "Linux"
 enableTypeIgnoreComments = true
diff --git a/source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py b/source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py
index 2f558183..4fcb58e4 100644
--- a/source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py
+++ b/source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py
@@ -202,7 +202,8 @@ class UniformPose2dCommandCfg(CommandTermCfg):
 
     @configclass
     class Ranges:
-        """Uniform distribution ranges for the position commands."""
+        """Uniform distribution ranges for the position commands.
+        2D位置命令的均匀分布范围配置类。"""
 
         pos_x: tuple[float, float] = MISSING
         """Range for the x position (in m)."""
@@ -222,8 +223,9 @@ class UniformPose2dCommandCfg(CommandTermCfg):
     goal_pose_visualizer_cfg: VisualizationMarkersCfg = GREEN_ARROW_X_MARKER_CFG.replace(
         prim_path="/Visuals/Command/pose_goal"
     )
-    """The configuration for the goal pose visualization marker. Defaults to GREEN_ARROW_X_MARKER_CFG."""
-
+    """The configuration for the goal pose visualization marker. Defaults to GREEN_ARROW_X_MARKER_CFG.
+    """
+    #
     # Set the scale of the visualization markers to (0.2, 0.2, 0.8)
     goal_pose_visualizer_cfg.markers["arrow"].scale = (0.2, 0.2, 0.8)
 
diff --git a/source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py b/source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py
index 76a07810..e82bba60 100644
--- a/source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py
+++ b/source/isaaclab/isaaclab/envs/mdp/commands/pose_2d_command.py
@@ -50,6 +50,13 @@ class UniformPose2dCommand(CommandTerm):
         # -- robot
         self.robot: Articulation = env.scene[cfg.asset_name]
 
+        print("=== Robot 对象的所有属性 ===")
+        print(dir(self.robot))
+    
+        # 查看 data 属性下的内容
+        print("\n=== Robot.data 的所有属性 ===")  
+        print(dir(self.robot.data))
+
         # crete buffers to store the command
         # -- commands: (x, y, z, heading)
         self.pos_command_w = torch.zeros(self.num_envs, 3, device=self.device)
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py
index 7a1fc12a..e8901d04 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py
@@ -26,6 +26,8 @@ if TYPE_CHECKING:
 def feet_air_time(
     env: ManagerBasedRLEnv, command_name: str, sensor_cfg: SceneEntityCfg, threshold: float
 ) -> torch.Tensor:
+    #reward的计算以来赖于环境和传感器的状态，怎么获得这些物理量，依赖ManagerBasedRLEnv和SceneEntityCfg
+    #除了env固定需要，其他的参数怎么传递进去的？答：通过RewardTermCfg的params参数传递进去，在配置文件中写好需要传递的参数有哪些
     """Reward long steps taken by the feet using L2-kernel.
 
     This function rewards the agent for taking steps that are longer than a threshold. This helps ensure
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/velocity_env_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/velocity_env_cfg.py
index 735333f9..606db37a 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/velocity_env_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/velocity_env_cfg.py
@@ -63,6 +63,7 @@ class MySceneCfg(InteractiveSceneCfg):
     # robots
     robot: ArticulationCfg = MISSING
     # sensors
+    #从仿真器的角度考虑传感器
     height_scanner = RayCasterCfg(
         prim_path="{ENV_REGEX_NS}/Robot/base",
         offset=RayCasterCfg.OffsetCfg(pos=(0.0, 0.0, 20.0)),
@@ -121,6 +122,8 @@ class ObservationsCfg:
         """Observations for policy group."""
 
         # observation terms (order preserved)
+        #不需要考虑顺序，按照格式来写的话，就会按照代码组织的顺序来存储和拼接
+        #对比legged gym的代码
         base_lin_vel = ObsTerm(func=mdp.base_lin_vel, noise=Unoise(n_min=-0.1, n_max=0.1))
         base_ang_vel = ObsTerm(func=mdp.base_ang_vel, noise=Unoise(n_min=-0.2, n_max=0.2))
         projected_gravity = ObsTerm(
@@ -234,6 +237,8 @@ class RewardsCfg:
     # -- task
     track_lin_vel_xy_exp = RewTerm(
         func=mdp.track_lin_vel_xy_exp, weight=1.0, params={"command_name": "base_velocity", "std": math.sqrt(0.25)}
+    #params的含义是传递给函数的参数，但是不会立刻执行函数，而是在计算奖励的时候才会调用函数
+
     )
     track_ang_vel_z_exp = RewTerm(
         func=mdp.track_ang_vel_z_exp, weight=0.5, params={"command_name": "base_velocity", "std": math.sqrt(0.25)}
@@ -249,6 +254,7 @@ class RewardsCfg:
         weight=0.125,
         params={
             "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*FOOT"),
+            #SceneEntityCfg
             "command_name": "base_velocity",
             "threshold": 0.5,
         },